game/client/sdShop.js

```diff
@@ -782,9 +775,49 @@ class sdShop
+			sdShop.options.push({ _class: 'sdFactionTools', type:8, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:9, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:10, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:11, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:12, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:13, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:14, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:15, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:16, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:17, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:18, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:19, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:20, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:21, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:22, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:23, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:24, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:25, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:26, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:27, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:28, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:29, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:30, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:31, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:32, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:33, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:34, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:35, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:36, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:37, _category:'Humanoid Spawner' });
+			sdShop.options.push({ _class: 'sdFactionTools', type:38, _category:'Humanoid Spawner' });
+
```

Comment:
I'm not sure if we need many lines of `sdShop.options.push({ _class: 'sdFactionTools', type: 1 2 3 exc, _category: 'Humanoid Spawner'  });`, could be reduced by making a for loop to iterating how many types there are.

Reply:
Umm I can't remove many type of spawning characters, but listed in sdShop.js is fixed.